controllers/playlist.controller.js

```diff
@@ -0,0 +1,168 @@
+const User = require("../models/user-model");
+const Playlist = require("../models/playlist-model");
+const { concat } = require("lodash");
+
+const getPlaylists = async (req, res) => {
+  const playlist = await Playlist.find({});
+  res.json({ success: true, playlist });
+};
+
+const findUserPlaylist = async (req, res, next, userId) => {
+  try {
+    let user = await User.findOne({ _id: userId });
+    if (!user) {
+      res.status(404).json({
+        success: false,
+        message: "Invalid user! Kindly register to continue",
+      });
+      throw Error("Invalid User");
+    }
+    let playlist = await Playlist.findOne({ userId });
+
+    if (!playlist) {
+      playlist = new Playlist({
+        userId,
+        playlists: [{ name: "Custom 1", videos: [], active: true }],
+      });
+      playlist = await playlist.save();
+    }
+    req.playlist = playlist;
+    next();
+  } catch (error) {
+    res.status(500).json({
+      success: false,
+      message: "Unable to retrive user's playlists",
+      errorMessage: error.message,
+    });
+  }
+};
+
+const getActivePlaylistItems = async (playlist) => {
+  playlist.playlists = playlist.playlists = playlist.playlists.filter(
+    (list) => list.active
+  );
```

Comment:
I am unable to understand this line  Kunal, can you please explain?

Reply:
This is a mistake. I have changed it